packages/teams-js/src/public/secondaryBrowser.ts

```diff
@@ -0,0 +1,50 @@
+import { sendAndHandleSdkError as sendAndHandleError } from '../internal/communication';
+import { ensureInitialized, isHostClientMobile } from '../internal/internalAPIs';
+import { isValidHttpsURL } from '../internal/utils';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { ErrorCode } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * Namespace to power up the in-app browser experiences in the Host App.
+ * For eg. opening a URL in the Host App inside a browser
+ *
+ * @beta
+ */
+export namespace secondaryBrowser {
+  /**
+   * Open a URL in the secondary browser aka in-app browser
+   *
+   * @param url Url to open in the browser
+   * @returns Promise that resolve to true if the URL successfully opens in the secondaryBrowser
+   * or throws an error {@link SdkError} incase of failure before starting navigation
+   *
+   * Pls note that any error that happens after starting navigation is delegated on external browsers to handle in their generic way
+   * @beta
+   */
+  export function open(url: URL): Promise<boolean> {
```

Comment:
Is there a reason this needs to be a `Promise<boolean>` rather than just `Promise<void>`? It seems like success or failure is communicated by whether the promise is resolved or rejected; is additionally having a `boolean` value duplicative?

Reply:
In my opinion for good Developer's experience its always helpful to provide an acknowledgement of the request if its completed successfully or failed. Promise<void> would not allow to convey the success side and only the failure side of the story. Let me know if that answers your question.

Reply:
I agree that it's necessary to indicate whether the request succeeded or failed, but I'm not sure I understand the distinction you are making. Isn't the point of either resolving or rejecting a promise to indicate success or failure? If you also include a boolean then you're indicating creating a matrix of return possibilities that seems unclear to developers. For example, what does it mean if the promise is resolved but the boolean is false? Is that a success or a failure? Or both?

Reply:
Discussed offline on this and sent an update.